src/crystal/scheduler.cr

```diff
@@ -234,7 +234,7 @@ class Crystal::Scheduler
           Thread.current.scheduler.enqueue worker_loop
           Thread.current
         else
-          Thread.new do
+          Thread.new(name: "mt-#{i}") do
```

Comment:
I really like (and will *definitely* use the feature), but the default names should be more descriptive. Personally I'd probably prefer the default to be something like `crystal-1`..`crystal-4` for the 4 threads. I'm including the main crystal in the list as there really is no crystal userland separation between them - that one happen to be a bit special under the hood is just an implementation detail from the point of the user, no? Does that leak through in any other situation?

Reply:
Or hmm. Do they behave differently if the pids are killed?

Reply:
I went with `mt-i` because it was simple & short. It could be `scheduler-i` or `mt-scheduler-i` but `crystal-i` isn't very informative.

We could maybe use the program name (I guess) but it would be `crystal-run-xyz.tmp-i` or `crystal-run-spec.tmp-i` for crystal run and crystal spec, but even `myapp-1` wouldn't be very informative.

Reply:
What I'm trying to address is the situation where someone goes into an unknown system and does `top` or similar to see what is going on. In that situation, `myapp-1` is a very big improvement to `mt-1`, as it gives a clear pointer of where to look and what's responsible for the thread. 

I agree that the runtime names of `crystal run` (and `crystal spec`, for that matter) isn't very helpful. Perhaps they can be improved upon? 

That said, naming is clearly something that can be discussed and improved upon after merging, so perhaps the question of thread and executable naming should be forked off to another issue? This PR is still a nice improvement on the status quo.

Reply:
Hum, maybe the program name would be helpful. We could match `/crystal-(?:run|spec)-(.*).tmp/` to only extract the program's name.

Let's note that I don't rename the main thread to make sure at least one thread keeps the program's name, but maybe we could have `myapp-0` then :thinking:

I'm gonna try that.

Reply:
It doesn't work: a thread name may only be 16 chars including the terminating NULL byte, and the name is truncated at 15 chars.

I think `CRYSTAL-MT-i` is the best default (uppercased to match GC).

Reply:
Fair enough, it can always be improved upon later if someone figures out something better.